packages/backend/src/ee/services/ai/agents/tests/utils/llmAsAJudge.ts

```diff
@@ -26,6 +26,17 @@ export type ContextRelevancyResponse = {
     reason: string;
 };
 
+export type LlmJudgeResult = {
+    scorerType: 'factuality' | 'jsonDiff' | 'contextRelevancy';
+    query: string;
+    response: string;
+    expectedAnswer?: string;
+    context?: string[];
+    result: FactualityResponse | JsonDiffResponse | ContextRelevancyResponse;
+    timestamp: string;
+    passed?: boolean;
+};
```

Comment:
```
type LlmJudgeResultBase = {
  query: string;
  response: string;
  expectedAnswer?: string;
  timestamp: string;
  passed?: boolean;
};

export type LlmJudgeResult =
  | (LlmJudgeResultBase & {
      scorerType: 'factuality';
      result: FactualityResponse;
    })
  | (LlmJudgeResultBase & {
      scorerType: 'jsonDiff';
      result: JsonDiffResponse;
    })
  | (LlmJudgeResultBase & {
      scorerType: 'contextRelevancy';
      context: string[];
      result: ContextRelevancyResponse;
    });
```

you can have nice union which will improve devx too to avoid impossible states + optionals